tests/foreman/api/test_user.py

```diff
@@ -4,70 +4,101 @@
 tested can be found here: http://theforeman.org/api/apidoc/v2/users.html
 
 """
-
-from fauxfactory import gen_string
 from nailgun import entities
-from random import randint
 from requests.exceptions import HTTPError
+from robottelo.datafactory import (
+    generate_strings_list,
+    valid_usernames_list,
+    valid_emails_list,
+    valid_user_data_list,
+    invalid_emails_list,
+    invalid_usernames_list,
+    invalid_names_list
+)
 from robottelo.decorators import tier1
 from robottelo.test import APITestCase
 
 
-def _user_attrs():
-    """Returns a tuple of dicts. Each dict contains user attributes.
-
-    Note: Supplying utf8 values for firstname, lastname or login is rejected
-    by Satellite
-    """
-    return (
-        {u'admin': False},
-        {u'admin': True},
-        {u'firstname': gen_string('alphanumeric', randint(1, 50))},
-        {u'firstname': gen_string('alpha', randint(1, 50))},
-        {u'firstname': gen_string('cjk', randint(1, 50))},
-        {u'firstname': gen_string('latin1', randint(1, 50))},
-        {u'firstname': gen_string('numeric', randint(1, 50))},
-        {u'lastname': gen_string('alphanumeric', randint(1, 50))},
-        {u'lastname': gen_string('alpha', randint(1, 50))},
-        {u'lastname': gen_string('cjk', randint(1, 50))},
-        {u'lastname': gen_string('latin1', randint(1, 50))},
-        {u'lastname': gen_string('numeric', randint(1, 50))},
-        {u'login': gen_string('alphanumeric', randint(1, 100))},
-        {u'login': gen_string('alpha', randint(1, 100))},
-        {u'login': gen_string('cjk', randint(1, 100))},
-        {u'login': gen_string('latin1', randint(1, 100))},
-        {u'login': gen_string('numeric', randint(1, 100))},
-    )
-
-
 class UserTestCase(APITestCase):
     """Tests for the ``users`` path."""
 
     @tier1
     def test_positive_create(self):
-        """Create a user with attributes ``attrs`` and delete it.
+        """Create User for all random variations of data
+        @Feature: User - Positive Create
```

Comment:
Please add a blank line above, this will follow the majority of docstrings styles.